tests/test_events.py

```diff
@@ -36,3 +36,80 @@ def on_validated(context):
     pyblish.util.validate()
 
     assert count["#"] == 1, count
+
+@with_setup(lib.setup_empty)
+def test_plugin_processed_event():
+    """pluginProcessed is emitted upon a plugin being processed, regardless of its success"""
+
+    class MyContextCollector(pyblish.api.ContextPlugin):
+        order = pyblish.api.CollectorOrder
+
+        def process(self, context):
+            context.create_instance("A")
+
+    class CheckInstancePass(pyblish.api.InstancePlugin):
+        order = pyblish.api.ValidatorOrder
+
+        def process(self, instance):
+            pass
+
+    class CheckInstanceFail(pyblish.api.InstancePlugin):
+        order = pyblish.api.ValidatorOrder
+
+        def process(self, instance):
+            raise Exception("Test Fail")
+
+    pyblish.api.register_plugin(MyContextCollector)
+    pyblish.api.register_plugin(CheckInstancePass)
+    pyblish.api.register_plugin(CheckInstanceFail)
+
+
+    count = {"#": 0}
+
+    def on_processed(result):
+        assert isinstance(result, dict)
+        count["#"] += 1
+
+    pyblish.api.register_callback("pluginProcessed", on_processed)
+    pyblish.util.publish()
+
+    assert count["#"] == 3, count
+
+@with_setup(lib.setup_empty)
+def test_plugin_failed_event():
+    """pluginFailed is emitted upon a plugin failing for any reason"""
+
+    class MyContextCollector(pyblish.api.ContextPlugin):
+        order = pyblish.api.CollectorOrder
+        def process(self, context):
+            context.create_instance("A")
+
+    class CheckInstancePass(pyblish.api.InstancePlugin):
+        order = pyblish.api.ValidatorOrder
+        def process(self, instance):
+            pass
+
+    class CheckInstanceFail(pyblish.api.InstancePlugin):
+        order = pyblish.api.ValidatorOrder
+        def process(self, instance):
+            raise Exception("Test Fail")
+
+    pyblish.api.register_plugin(MyContextCollector)
+    pyblish.api.register_plugin(CheckInstancePass)
+    pyblish.api.register_plugin(CheckInstanceFail)
+
+    count = {"#": 0}
+
+    def on_failed(plugin, context, instance, error):
+        #todo: add further checks for the other incoming args
+        #assert isinstance(instance, CheckInstanceFailRaise)
```

Comment:
Ah, yes the plug-in you're getting is probably not an instance, but the class itself. Try something like `issubclass(plugin, pyblish.api.InstancePlugin)` or simply `plugin == CheckInstanceFail`.